### DISCUSSION TITLE
[Idea] Generic Component

### DISCUSSION THREAD

**Discussion by Cahllagerfeld** [spences10/sveltekit-embed#178] [Ideas]
To make the package even more flexible, a generic component might be a possibility.
What I mean by this, is an iframe wrapped inside the `Observercomponent`, taking in any props with `$$restProps`.
Also a `<slot/>` might be added so users can put their own markup inside the Observer, in case there need to be further customizations.

```svelte
// Quick pseudocode snippet
<Observer>
    <iframe {...$$restProps}/>
    <slot/>
</Observer>

```

Let me know what you think about it 🤔 🛹

**Comment by spences10** [comment]
Yeah, great idea @Cahllagerfeld!

If you're ok doing that, let me know how it goes.

Do you have something in mind for what to use it for?

As an aside, the example sheet should probably have some work done with usage etc, but I _really_ don't want to use Storybook, if you look back at the commits this did use Storybook but I abandoned it because it was a pain to use 😅 I think that is another discussion though! 😊

**Reply by Cahllagerfeld** [reply]
It was just an idea, with the use-case for users wanting to embed a service the package isn't providing out-of-the-box.
With the generic approach, the package might be used for potentially every embed out there.
I think it's discussable if the package is needed in that case 🤔 The main reason to use the generic component instead of a regular iframe might be the observer component.

I also didn't make that good experience with Sveltekit and Storybook 😅

**Reply by spences10** [reply]
Agreed, the component could be the 'catch all' if there's nothing provided with the package.
